### DISCUSSION TITLE
Stop brew command from terminating a sudo session

### DISCUSSION THREAD

**Discussion by i0ntempest** [Homebrew/discussions#5806] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: ad356d365837a691a52dfec32b3fb5f282f3aefc
Last commit: 5 days ago
Branch: stable
Core tap HEAD: 34b2b819b4f857ea2c5d81e0b6472ad90c22bd7c
Core tap last commit: 2 hours ago
Core cask tap HEAD: 33a0f29066d8cdfb220fb19884aa4976b1194fb6
Core cask tap last commit: 17 minutes ago
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.q2wsZnCJpj/org.macports:0
HOMEBREW_EDITOR: /opt/local/bin/nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 24
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 24-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.2-arm64
CLT: 16.2.0.0.1.1733547573
Xcode: 16.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  cheatsheet
  edex-ui
  mono-mdk-for-visual-studio
  paragon-vmdk-mounter
  puush
  quicklookapk
  qv2ray
  save-hollywood
  visual-studio
```

### Description of issue

I have a script that automates updating all packages on my system. But because brew command now terminates the sudo session every time it runs, the script asks my password multiple times, whereas before it only ask once. How to get around this?
And yes I am aware the askpass thing and I've tried it before, and I decided that I don't like it.

**Comment by SMillerDev** [comment]
> How to get around this?

There is only one way I know to get around this.
> And yes I am aware the askpass thing and I've tried it before, and I decided that I don't like it.

It's this one

**Reply by i0ntempest** [reply]
IMHO there should be a env variable to turn this behavior off. What's the rationale	for this behavior tho? I can't really see any and it just complicates my scripting.

**Reply by gromgit** [reply]
> What's the rationale for this behavior tho?

https://brew.sh/2024/07/30/homebrew-security-audit/, specifically item 12 "Formula privilege escalation through sudo".

**Comment by gromgit** [comment, marked as answer]
If your complaint is that Homebrew resets sudo credentials so that commands _after_ `brew` now prompt for a password, you can simply use `script` to isolate the credential reset. Since you're on macOS, I believe `script /dev/null brew ...` should do the trick.

**Reply by i0ntempest** [reply]
Works, thanks!

**Reply by loscrossos** [reply]
@gromgit 
THANKS FROM THE FUTURE!!
